crates/rooch-framework/sources/session_key.move

```diff
@@ -0,0 +1,144 @@
+        authentication_key: vector<u8>,
+        scheme: u64,
+        scopes: vector<SessionScope>,
+        /// The session key's expiration time period, in seconds, 0 means never expired
+        expiration_time: u64,
+        /// The session key's last active time
+        last_active_time: u64,
+        /// The session key's max inactive time period, in seconds
+        max_inactive_interval: u64,
+    }
+
+    struct SessionKeys has key {
+        keys: Table<vector<u8>, SessionKey>,
+    }
+
+    public fun is_expired(_ctx: &StorageContext, _session_key: &SessionKey) : bool {
+        //TODO check the session key is expired or not after the timestamp is supported
+        return false
+    }
+
+    public fun exists_session_key(ctx: &StorageContext, account_address: address, authentication_key: vector<u8>) : bool {
+        option::is_some(&get_session_key(ctx, account_address, authentication_key))
+    }
+
+    /// Get the session key of the account_address by the authentication key
+    public fun get_session_key(ctx: &StorageContext, account_address: address, authentication_key: vector<u8>) : Option<SessionKey> {
+        if (!account_storage::global_exists<SessionKeys>(ctx, account_address)){
+            return option::none()
+        };
+        let session_keys = account_storage::global_borrow<SessionKeys>(ctx, account_address);
+        if (!table::contains(&session_keys.keys, authentication_key)){
+            return option::none()
+        }else{
+            option::some(*table::borrow(&session_keys.keys, authentication_key))
+        }
+    }
+
+    public fun create_session_key(ctx: &mut StorageContext, sender: &signer, authentication_key: vector<u8>, scheme: u64, scopes: vector<SessionScope>, expiration_time: u64, max_inactive_interval: u64) {
+        //Can not create new session key by the other session key
+        assert!(!auth_validator::is_validate_via_session_key(ctx), error::permission_denied(ESessionKeyCreatePermissionDenied));
+        let sender_addr = signer::address_of(sender);
+        assert!(!exists_session_key(ctx, sender_addr, authentication_key), error::already_exists(ESessionKeyAlreadyExists));
+
+        let session_key = SessionKey {
+            authentication_key: authentication_key,
+            scheme: scheme,
+            scopes: scopes,
+            expiration_time: expiration_time,
+            //TODO set the last active time to now
+            last_active_time: 0,
+            max_inactive_interval: max_inactive_interval,
+        };
+        if (!account_storage::global_exists<SessionKeys>(ctx, sender_addr)){
+            let keys = table::new<vector<u8>, SessionKey>(storage_context::tx_context_mut(ctx));
+            account_storage::global_move_to<SessionKeys>(ctx, sender, SessionKeys{keys});
+        };
+
+        let session_keys = account_storage::global_borrow_mut<SessionKeys>(ctx, sender_addr);
+        table::add(&mut session_keys.keys, authentication_key, session_key);
```

Comment:
If the auth key rotated is implemented later, the corresponding sessionkey must be invalidated after being rotated, which requires a contract method